### PR TITLE
Corrected MCDU flight phase

### PR DESCRIPTION
**Pull request by Aryeng**
**Fixes**
- MCDU is by asobo default put into TAKEOFF flight phase after loading into sim. This fix sets flight phase to PREFLIGHT after loading.
- Adding the proper PREFLIGHT phase also fixes the enroute flight time estimation (not a fix for actual estimation) in the flightplanpage prior to flying.

**Summary of Changes**
- Added line in init that puts mcdu in the correct PREFLIGHT flight phase.
- Added necessary checks for the mcdu to change from PREFLIGHT to TAXI and then TAKEOFF. These checks along with default asobo checks also switch to correct flight phase if spawning on runway or in air. 

Notes: Performance page will show TAKEOFF in white text at gate, during taxi and at lineup/on runway until takeoff power is set (flex or toga). Then it switches to green text when flightphase has been changed to TAKEOFF. 

With this fix, the mcdu flightplan page does not show UTC estimated time of arrival during preflight and taxi (like real life afaik). It shows calculated flight time and you can then add a UTC constraint (not implemented yet) to get UTC estimated times according to scheduled departure/takeoff time (utc constraint input). Flight plan page returns to UTC estimated time of arrival (which seems pretty broken by default) once takeoff phase has been activated.

**Additional context**
First fix of many to implement basic fuel calculation and more accurate flight time calculation.

Currently, this fix is not applied to the current "turnaround-code snippet", so don't expect the flightplan page to show correct preflight information (non-UTC times) at turnaround. Does turnarounds even work right now? I haven't tested it lately. I have an idea how to fix the turnaround code if it's not working. It is very easy to include this fix into the current turnaround code, but I didnt include it because I am unsure if the turnaround code is working as intended.

Discord username (if different from GitHub): Ryeng#0675

### CODE REVIEW

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js [hunk 1]
```diff
@@ -85,6 +85,7 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
 	        }
         this.electricity = this.querySelector("#Electricity");
         this.climbTransitionGroundAltitude = null;
+		this.currentFlightPhase = FlightPhase.FLIGHT_PHASE_PREFLIGHT;
```
Comment (Nolin7777) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js:88]: tabs

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js [hunk 2]
```diff
@@ -889,12 +890,18 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
         const rightThrottleDetent = Simplane.getEngineThrottleMode(1);
         const highestThrottleDetent = (leftThrottleDetent >= rightThrottleDetent) ? leftThrottleDetent : rightThrottleDetent;
 
+		//Changes from PREFLIGHT to TAXI PHASE after engine startup
```
Comment (Nolin7777) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js:893]: tabs

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js (hunk 1 above)
Comment (Aryeng) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js:88]: idk why it looks like that on git. was normal in my file before uploading.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js [hunk 3]
```diff
@@ -889,12 +890,18 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
...
+			}
+		}
 
-
-        //End preflight when takeoff power is applied and engines are running
+        //Changes from TAXI to TAKEOFF PHASE and completes "preflight" when takeoff power is applied and engines are running
         if (this.currentFlightPhase <= 2) {
             if ((highestThrottleDetent == ThrottleMode.TOGA || highestThrottleDetent == ThrottleMode.FLEX_MCT) && SimVar.GetSimVarValue("ENG N1 RPM:1", "Percent") > 15 && SimVar.GetSimVarValue("ENG N1 RPM:2", "Percent") > 15) {
                 SimVar.SetSimVarValue("L:A32NX_Preflight_Complete", "Bool", 1);
+				this.currentFlightPhase = FlightPhase.FLIGHT_PHASE_TAKEOFF;
```
Comment (Nolin7777) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js:904]: brackets

**Comment by NathanInnes** [comment]
Please include CHANGELOG with the title of this PR and your name. It's located under .github 

If you need assistance I am on discord to help

**Comment by Aryeng** [comment]
> 
> 
> Please include CHANGELOG with the title of this PR and your name. It's located under .github
> 
> If you need assistance I am on discord to help

Not sure if I added the changelog correctly. whats your username on discord?

**Comment by NathanInnes** [comment]
> > Please include CHANGELOG with the title of this PR and your name. It's located under .github
> > If you need assistance I am on discord to help
> 
> Not sure if I added the changelog correctly. whats your username on discord?

Sabes#8668

File: .github/CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,7 @@
 <!--  1. [Changed Area] Title of changes - @github username (Name)  -->
 ## Changes from 2020/09 to 2020/10
 
+1. [CDU] Corrected MCDU flight phase - @aryeng (ryeng)
 1. [General] Add CHANGELOG.md - @nathaninnes (Nathan Innes)
 2. [ECAM] Improved Upper ECAM Visuals - @wpine215 (Iceman)
```
Comment (NathanInnes) [inline .github/CHANGELOG.md:11]: Order
```suggestion
1. [General] Add CHANGELOG.md - @nathaninnes (Nathan Innes)
2. [ECAM] Improved Upper ECAM Visuals - @wpine215 (Iceman)
1. [CDU] Corrected MCDU flight phase - @aryeng (ryeng)
```

File: .github/CHANGELOG.md [hunk 5]
```diff
@@ -7,4 +7,8 @@
 
 1. [General] Add CHANGELOG.md - @nathaninnes (Nathan Innes)
 2. [ECAM] Improved Upper ECAM Visuals - @wpine215 (Iceman)
+1. [CDU] Corrected MCDU flight phase - @aryeng (ryeng)
+<<<<<<< HEAD
 
+=======
+>>>>>>> 368676ef45c529b456b8fe077eea78bfde505dc4
```
Comment (Benjozork) [inline .github/CHANGELOG.md:14]: You have unresolved merge conflicts in this file.

Comment (NathanInnes) [inline .github/CHANGELOG.md:14]:
```suggestion
```

Comment (Aryeng) [reply .github/CHANGELOG.md]: I commited your suggested, however still conflicting. Nathan told me yesterday to not worry about the conflict in changelog.

Comment (Aryeng) [reply .github/CHANGELOG.md]: Updated the changelog. Seems to have cleared the condlict.

**Comment by lousybyte** [comment]
Something ain't right about how this was submitted, **"Files changed: 112"**.

**Comment by Aryeng** [comment]
those came after updating the vanilla files to 1.9.3

**Comment by lukecologne** [comment]
You need to either rebase onto master or merge master into your branch and resolve all conflicts

**Comment by Aryeng** [comment]
@lukecologne 
I will do another merge soon.

**Comment by Aryeng** [comment]
Re-merged latest master. Tbh, I might just pull this PR and just post my whole fuel prediction in it's entirety. Splitting it up into smaller PRs is gonna take way too long based on the time this PR has been sitting here..

**Comment by lousybyte** [comment]
Still not quite right, as much as we welcome contributions, merging this would overwrite about 150 files. If you are having issues using git, you can just open a new PR from a fresh fork from master with your changes and mention this one in the description.

For future reference: [https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/syncing-a-fork](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/syncing-a-fork)

**Pull request closed without merging**